### PR TITLE
Rwa 230

### PR DESCRIPTION
**Pull request by indvamshi**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RWA-230

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

**Review by lewis150193 — APPROVED** [review]

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskForSearchCompletionControllerTest.java [hunk 1]
```diff
@@ -142,5 +150,45 @@ public void should_return_a_200_and_when_performing_search_when_caseId_correct_e
         common.cleanUpTask(taskId, REASON_COMPLETED);
     }
 
+    @Test
```
Comment (BenNzewi) [inline src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskForSearchCompletionControllerTest.java:153]: Excellent scenario coverage.

One small suggestion; So we adhere to the test pyramid and lean coverage in FT(One positive test case + 1 other edge case), could you move "should_return_a_500_and_when_performing_search_when_caseType_is_incorrect()" under integration test?

Comment (indvamshi) [reply src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskForSearchCompletionControllerTest.java:153]: The reason for adding jurisdiction/casetype in FT is it actually calls camunda endpoint and checks for both the keys. But in IT, we are mocking to respond to 500 errorCode so the key is irrelevant.
